src/DripsDeployer.sol

```diff
@@ -0,0 +1,412 @@
+    address public immutable dripsDeployer;
+    bytes32 public immutable moduleSalt;
+
+    constructor(address dripsDeployer_, bytes32 moduleSalt_) {
+        require(
+            address(this) == Create3.computeAddress(moduleSalt_, dripsDeployer_),
+            "Invalid module deployment salt"
+        );
+        dripsDeployer = dripsDeployer_;
+        moduleSalt = moduleSalt_;
+    }
+
+    function args() public view virtual returns (bytes memory);
+
+    function _moduleAddress(bytes32 salt) internal view returns (address addr) {
+        return Create3.computeAddress(salt, dripsDeployer);
+    }
+
+    modifier onlyModule(bytes32 salt) {
+        require(msg.sender == _moduleAddress(bytes32(salt)));
+        _;
+    }
+}
+
+abstract contract ContractDeployerModule is BaseModule {
+    bytes32 public immutable salt = "deployment";
+
+    function deployment() public view returns (address) {
+        return Create3.computeAddress(salt);
+    }
+
+    function deploymentArgs() public view virtual returns (bytes memory);
+
+    function _deployContract(bytes memory creationCode) internal {
+        Create3.deploy(0, salt, abi.encodePacked(creationCode, deploymentArgs()));
+    }
+}
+
+abstract contract ProxyDeployerModule is BaseModule {
+    bytes32 public immutable proxySalt = "proxy";
+    address public proxyAdmin;
+    address public logic;
+
+    function proxy() public view returns (address) {
+        return Create3.computeAddress(proxySalt);
+    }
+
+    function proxyArgs() public view returns (bytes memory) {
+        return abi.encode(logic, proxyAdmin);
+    }
+
+    function logicArgs() public view virtual returns (bytes memory);
+
+    // slither-disable-next-line reentrancy-benign
+    function _deployProxy(address proxyAdmin_, bytes memory logicCreationCode) internal {
+        // Deploy logic
+        address logic_;
+        bytes memory logicInitCode = abi.encodePacked(logicCreationCode, logicArgs());
+        // slither-disable-next-line assembly
+        assembly ("memory-safe") {
+            logic_ := create(0, add(logicInitCode, 32), mload(logicInitCode))
+        }
+        require(logic_ != address(0), "Logic deployment failed");
+        logic = logic_;
+        // Deploy proxy
+        proxyAdmin = proxyAdmin_;
+        // slither-disable-next-line too-many-digits
+        bytes memory proxyInitCode = abi.encodePacked(type(ManagedProxy).creationCode, proxyArgs());
+        Create3.deploy(0, proxySalt, proxyInitCode);
+    }
+}
+
+abstract contract DripsDependentModule is BaseModule {
+    // slither-disable-next-line naming-convention
+    bytes32 internal immutable _dripsModuleSalt = "Drips";
+
+    function _dripsModule() internal view returns (DripsModule) {
+        address module = _moduleAddress(_dripsModuleSalt);
+        require(Address.isContract(module), "Drips module not deployed");
+        return DripsModule(module);
+    }
+}
+
+contract DripsModule is DripsDependentModule, ProxyDeployerModule {
```

Comment:
```suggestion
contract DripsModule is ProxyDeployerModule {
```

Reply:
I don't think the `DripsModule` needs this dependency

Reply:
Ah, I see you not using the method but you need the `_dripsModuleSalt` from the `DripsDependentModule`